cored/src/commonMain/kotlin/com/github/kittinunf/cored/engine/HashEngine.kt

```diff
@@ -51,4 +55,13 @@ internal class HashEngine<S : Any, A : Any>(
     override fun removeMiddleware(middleware: AnyMiddleware<S>): Boolean {
         error("This engine does not support this removeMiddleware without key, please use fun removeMiddleware(key: Any, middleware: AnyMiddleware<S>) instead")
     }
+
+    override fun addReducer(actionReducer: ActionReducer<S, Any>) {
+        val (key, reducer) = actionReducer
+        reducerMap.put(key, reducer)
+    }
+
+    override fun removeReducer(key: Any): Boolean {
+        return reducerMap.remove(key) != null
+    }
```

Comment:
I would suggest to have method with same parameter type as for `addReducer()`:
```
override fun removeReducer(actionReducer: ActionReducer<S, Any>): Boolean { ... }
```
Assuming user of the API might not know at first look what's the `key` (as he is not passing the explicit `key` when adding reducer) and pass `ActionReducer` (as the type is `Any`, it's possible) to this method, which will not work.

Reply:
Same like for middleware :)

Reply:
kk!